concordium-std/src/traits.rs

```diff
@@ -133,33 +135,50 @@ where
     fn get_key(&self) -> Vec<u8>;
 }
 
-pub trait HasContractStateLL {
-    type ContractStateData;
-    type EntryType: HasContractStateEntry;
-    type IterType: Iterator<Item = Self::EntryType>;
+pub trait HasContractStateLL: Debug {
```

Comment:
Why the debug supertrait? 

Code size is an issue, so we should not require things in smart contracts that are not crucial.